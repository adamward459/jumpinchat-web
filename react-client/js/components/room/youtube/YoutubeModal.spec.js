import React from 'react';
import { shallow } from 'enzyme';
import { YoutubeModal } from './YoutubeModal.react';

describe('<YoutubeModal />', () => {
  let props;
  beforeEach(() => {
    props = {
      isOpen: true,
      error: null,
      resultsLoading: false,
      playlist: [{
        _id: 'foo',
        mediaType: 'yt',
        startedBy: {
          userId: 'foo',
          username: 'foobar',
          pic: 'foo.jpg',
        },
        duration: 123,
        title: 'foo',
        description: 'description',
        channelId: 'abc',
        link: 'https://youtu.be/foo',
      }],
    };
  });

  it('should show playlist if search not active', () => {
    const wrapper = shallow(<YoutubeModal {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('should not show playlist if search loading', () => {
    props.resultsLoading = true;
    const wrapper = shallow(<YoutubeModal {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('should not show playlist if search has results', () => {
    props.results = [{

    }];
    const wrapper = shallow(<YoutubeModal {...props} />);
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
