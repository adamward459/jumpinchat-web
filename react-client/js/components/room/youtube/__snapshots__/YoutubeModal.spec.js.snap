// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YoutubeModal /> should not show playlist if search has results 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal__Window"
  closeTimeoutMS={0}
  contentLabel="Search youtube videos"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="modal"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <div
    className="modal__Header"
  >
    Search for YouTube videos
    <button
      className="modal__Button modal__Button-close"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-times"
      />
    </button>
  </div>
  <div
    className="modal__Body"
  >
    <YoutubeSearchModal
      error={null}
      results={
        Array [
          Object {},
        ]
      }
      resultsLoading={false}
    />
  </div>
</Modal>
`;

exports[`<YoutubeModal /> should not show playlist if search loading 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal__Window"
  closeTimeoutMS={0}
  contentLabel="Search youtube videos"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="modal"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <div
    className="modal__Header"
  >
    Search for YouTube videos
    <button
      className="modal__Button modal__Button-close"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-times"
      />
    </button>
  </div>
  <div
    className="modal__Body"
  >
    <YoutubeSearchModal
      error={null}
      results={Array []}
      resultsLoading={true}
    />
  </div>
</Modal>
`;

exports[`<YoutubeModal /> should show playlist if search not active 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal__Window"
  closeTimeoutMS={0}
  contentLabel="Search youtube videos"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="modal"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
>
  <div
    className="modal__Header"
  >
    Search for YouTube videos
    <button
      className="modal__Button modal__Button-close"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-times"
      />
    </button>
  </div>
  <div
    className="modal__Body"
  >
    <YoutubeSearchModal
      error={null}
      results={Array []}
      resultsLoading={false}
    />
    <YoutubePlaylist
      error={null}
      list={
        Array [
          Object {
            "_id": "foo",
            "channelId": "abc",
            "description": "description",
            "duration": 123,
            "link": "https://youtu.be/foo",
            "mediaType": "yt",
            "startedBy": Object {
              "pic": "foo.jpg",
              "userId": "foo",
              "username": "foobar",
            },
            "title": "foo",
          },
        ]
      }
      onRemoveItem={[Function]}
    />
  </div>
</Modal>
`;
